### PR TITLE
feat: [CDS-72777]: AWS CDK bootstrap step skeleton

### PR DESCRIPTION
**Pull request by tmacari-harness**
## Describe your changes
feat: [CDS-72777]: AWS CDK bootstrap step skeleton

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[CDS-72777]: https://harness.atlassian.net/browse/CDS-72777?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50193)

### CODE REVIEW

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger smartchecks

**Comment by sahilhindwani-harness** [comment]
*[`125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/awscdk/AwsCdkBootstrapStepPlanCreator.java` line 44 at r1](https://reviewable.io/reviews/harness/harness-core/50193#-NZzbePU9GbCPe-wmNP7:-NZzbePU9GbCPe-wmNP8:b-q8mkh8) ([raw file](https://github.com/harness/harness-core/blob/eb5d2a85bc38969da57d1f20154c1149e496789d/125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/awscdk/AwsCdkBootstrapStepPlanCreator.java#L44)):*
> ```Java
>   protected StepParameters getStepParameters(PlanCreationContext ctx, AwsCdkBootstrapStepNode stepNode) {
>     final StepParameters stepParameters = super.getStepParameters(ctx, stepNode);
>     AwsCdkBootstrapStepParameters awsCdkBootstrapStepParameters =
> ```

Is this required? Could you please check if we are able to make it work without this?

**Comment by sahilhindwani-harness** [comment]
*[`125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json` line 1540 at r1](https://reviewable.io/reviews/harness/harness-core/50193#-NZzcbYn2jWbWOQnk0cV:-NZzcbYn2jWbWOQnk0cW:b-8kcv64) ([raw file](https://github.com/harness/harness-core/blob/eb5d2a85bc38969da57d1f20154c1149e496789d/125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json#L1540)):*
> ```JSON
>       ],
>       "$schema": "http://json-schema.org/draft-07/schema#"
>     },
> ```

Please update static schema as well
[https://harness.atlassian.net/wiki/spaces/PIPE/pages/21420410346/Developer+Guide+for+Static+Schema](https://harness.atlassian.net/wiki/spaces/PIPE/pages/21420410346/Developer+Guide+for+Static+Schema)

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]
> Reviewed 24 of 24 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50193)* status: Some of the checks are too old. 24 file(s) to review. 1 change requests. 0 of 2 approvals obtained.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/provision/awscdk/AwsCdkBootstrapStepInfo.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        .envVariables(getEnvVariables())
+        .delegateSelectors(this.getDelegateSelectors())
+        .exportTemplate(this.getExportTemplate())
+        .build();
+  }
+
+  @Override
+  public ParameterField<List<TaskSelectorYaml>> fetchDelegateSelectors() {
+    return getDelegateSelectors();
+  }
```
Comment (lovish1999) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/provision/awscdk/AwsCdkBootstrapStepInfo.java:92]: I think we need to add this class in _io/harness/cdng/pipeline/CDStepInfo.java_ also..

Comment (tmacari-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/provision/awscdk/AwsCdkBootstrapStepInfo.java:92]: Done

**Review by tmacari-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50193)* status: 26 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/awscdk/AwsCdkBootstrapStepPlanCreator.java` line 44 at r1](https://reviewable.io/reviews/harness/harness-core/50193#-NZzbePU9GbCPe-wmNP7:-N_2ArAG0Z7pLuaPvwAV:b7u8byf) ([raw file](https://github.com/harness/harness-core/blob/eb5d2a85bc38969da57d1f20154c1149e496789d/125-cd-nextgen/src/main/java/io/harness/cdng/creator/plan/steps/awscdk/AwsCdkBootstrapStepPlanCreator.java#L44)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

Is this required? Could you please check if we are able to make it work without this?
</blockquote></details>

We need step parameters in step. The same is done for other steps.
___
*[`125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json` line 1540 at r1](https://reviewable.io/reviews/harness/harness-core/50193#-NZzcbYn2jWbWOQnk0cV:-N_2BHJfEhw0hxIw-Z86:b-d7bhn0) ([raw file](https://github.com/harness/harness-core/blob/eb5d2a85bc38969da57d1f20154c1149e496789d/125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json#L1540)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

Please update static schema as well
[https://harness.atlassian.net/wiki/spaces/PIPE/pages/21420410346/Developer+Guide+for+Static+Schema](https://harness.atlassian.net/wiki/spaces/PIPE/pages/21420410346/Developer+Guide+for+Static+Schema)
</blockquote></details>

will do

**Comment by tmacari-harness** [comment]
trigger feature-build

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger ut9

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger feature-build

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger feature-build

**Review by tmacari-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50193)* status: Some of the checks are too old. 30 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json` line 1540 at r1](https://reviewable.io/reviews/harness/harness-core/50193#-NZzcbYn2jWbWOQnk0cV:-N_XR22t-bCFdf8lcyyz:bqf5mez) ([raw file](https://github.com/harness/harness-core/blob/eb5d2a85bc38969da57d1f20154c1149e496789d/125-cd-nextgen/src/test/resources/schema/DeploymentStage/all.json#L1540)):*
<details><summary><i>Previously, tmacari-harness (Tudor Macari) wrote…</i></summary><blockquote>

will do
</blockquote></details>

https://github.com/harness/harness-schema/pull/64

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by sahilhindwani-harness** [comment]
*[`125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/AwsCdkBootstrapPluginInfoProvider.java` line 97 at r8](https://reviewable.io/reviews/harness/harness-core/50193#-N_YceFhCIZOKNAEn4-_:-N_YceFhCIZOKNAEn4-a:bzbx2gy) ([raw file](https://github.com/harness/harness-core/blob/dc8bb758bb8f4fb5b2bc45057d73eaa6b2d0677f/125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/AwsCdkBootstrapPluginInfoProvider.java#L97)):*
> ```Java
> 
>     environmentVariablesMap.put(
>         "PLUGIN_AWS_CDK_APP_PATH", getParameterFieldValue(awsCdkBootstrapStepInfo.getAppPath()));
> ```

Could you please create a environment variable constant file inside the cdk package and use it here.
It would help know which environment variables are being used inside this step or cdk as a general

**Review by sahilhindwani-harness — COMMENTED** [review]
> Reviewed 1 of 6 files at r2, 2 of 11 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, 18 of 19 files at r7, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50193)* status: 30 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained.

**Review by sahilhindwani-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50193)* status: 30 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by lovish1999 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50193)* status: 30 file(s) to review. 1 change requests. 1 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/provision/awscdk/AwsCdkBootstrapStepInfo.java` line 92 at r1](https://reviewable.io/reviews/harness/harness-core/50193#gh-1258314613:-N_YhM2f9FDJQezWEnxa:b2ni9b3) ([raw file](https://github.com/harness/harness-core/blob/eb5d2a85bc38969da57d1f20154c1149e496789d/125-cd-nextgen/src/main/java/io/harness/cdng/provision/awscdk/AwsCdkBootstrapStepInfo.java#L92)):*
<details><summary><i>Previously, tmacari-harness (Tudor Macari) wrote…</i></summary><blockquote>

Done
</blockquote></details>

ok

**Review by lovish1999 — COMMENTED** [review]
> Reviewed 6 of 24 files at r1, 1 of 6 files at r2, 2 of 11 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, 18 of 19 files at r7, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50193)* status: 1 change requests. 1 of 2 approvals obtained.

**Review by lovish1999 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50193)* status: :shipit: complete! ✓

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger ut0

**Comment by tmacari-harness** [comment]
trigger ut2

**Comment by tmacari-harness** [comment]
trigger ut6

**Comment by tmacari-harness** [comment]
trigger ut7

**Comment by tmacari-harness** [comment]
trigger ut0

**Comment by tmacari-harness** [comment]
trigger ut2

**Comment by tmacari-harness** [comment]
trigger ut7

**Comment by tmacari-harness** [comment]
trigger ut2

**Comment by tmacari-harness** [comment]
trigger ut7

**Review by yogesh-chauhan — APPROVED** [review]

**Pull request merged**